web/src/lib/components/asset-viewer/actions/keep-this-delete-others.svelte

```diff
@@ -0,0 +1,22 @@
+<script lang="ts">
+  import MenuOption from '$lib/components/shared-components/context-menu/menu-option.svelte';
+  import { AssetAction } from '$lib/constants';
+  import { keepThisDeleteOthers } from '$lib/utils/asset-utils';
+  import type { AssetResponseDto, StackResponseDto } from '@immich/sdk';
+  import { mdiPinOutline } from '@mdi/js';
+  import type { OnAction } from './action';
+  import { t } from 'svelte-i18n';
+
+  export let stack: StackResponseDto;
+  export let asset: AssetResponseDto;
+  export let onAction: OnAction;
+
+  const handleKeepThisDeleteOthers = async () => {
+    const keptAsset = await keepThisDeleteOthers(asset.id, stack.id);
```

Comment:
You already have the asset and stack here, no? Just change the `keepThisDeleteOthers` to accept those instead of getting the IDs and querying for them again.

Reply:
Function signature has been updated to accept the Asset & Stack objects instead of ids. Originally implemented this way as passing ids to the functions seemed to be the most common way elsewhere, but I completely agree that passing in the objects themselves is much more efficient.